azavea/ashlar-blueprint
# Add data collector components to the schema editor

## Overview

This PR pulls the data collector components out of DRIVER and adds them in as new views to the backend schema editor app.

### Checklist


### Demo

Log in to the schema editor at `localhost:9000`:

![schema-editor](https://user-images.githubusercontent.com/14170650/41681299-35c75e3a-74a2-11e8-8bcb-839ca00eec9d.png)

Select **Add a new record type** from the sidebar to create a RecordType:

![add-recordtype](https://user-images.githubusercontent.com/14170650/41681317-425fa18e-74a2-11e8-9619-c7c7db77248c.png)

Create your record type, then select **Add a new record** from its section in the sidebar:

![incident-options](https://user-images.githubusercontent.com/14170650/41681343-594ac1a8-74a2-11e8-95be-9b50c94f4a9a.png)

Add some information to your record:

![new-incident](https://user-images.githubusercontent.com/14170650/41681359-62cb7ff6-74a2-11e8-8d88-02afa2d25408.png)

Once you've created your record, select **View records** from the record type section in the sidebar to see a list of records and view more information about them:

![incident-details](https://user-images.githubusercontent.com/14170650/41681387-76bcda6e-74a2-11e8-8936-fadfd28a9696.png)

## Testing Instructions

Right now, the best way to test this PR is to spin it up and try all the functions to make sure they work. There are corresponding tests for a lot of these components in the DRIVER repo; I spent about two hours trying to get them to work with this version of the app, but they kept failing for esoteric reasons (e.g. the builtin `$httpBackend` mock was not available) and at a certain point I cut my losses and gave up on them since we'll be rewriting a lot of this code anyway.

That being said, if these tests are something that we should be include, I'd be glad to pair with someone on figuring out why they're not getting set up properly.
